File: Codeforces/119/a/a.cpp
#include <iostream>
#include <algorithm>

using namespace std;

int main() {
  int n, a, b, c;
  cin >> n >> a >> b >> c;
  int ret = 0;
  for (int i = 0; i <= n; ++i) {
    if (a * i > n) break;
    for (int j = 0; j <= n; ++j) {
      if (a * i + b * j > n) break;
      int left = n - (a * i + b * j);
      if (left % c == 0) {
        int k = left / c;
        ret = max(ret, i + j + k);
      }
    }
  }
  cout << ret << endl;
  return 0;
}

File: Codeforces/119/a/a.py
import sys

n, a, b, c = map(int, sys.stdin.readline().split())

ret = 0
for i in xrange(0, n+1):
    if a * i > n:
        break
    for j in xrange(0, n+1):
        if a * i + b * j > n:
            break
        left = n - (a * i + b * j)
        if left % c == 0:
            k = left // c
        else:
            continue
        ret = max(ret, i + j + k)

print ret
